lib/spack/llnl/util/cpu/microarchitectures.json

```diff
@@ -1161,6 +1161,97 @@
         }
       }
     },
+    "thunderx2": {
+      "from": "aarch64",
+      "vendor": "0x43",
+      "features": [
+        "fp",
+        "asimd",
+        "evtstrm",
+        "aes",
+        "pmull",
+        "sha1",
+        "sha2",
+        "crc32",
+        "atomics",
+        "cpuid",
+        "asimdrdm"
+      ],
+      "compilers": {
+        "gcc": [
+          {
+            "versions": "4.8:4.8.9",
+            "flags": "-march=armv8-a"
+          },
+          {
+            "versions": "4.9:5.9",
+            "flags": "-march=armv8-a+crc+crypto"
+          },
+          {
+            "versions": "6:6.9",
+            "flags": "-march=armv8.1-a+crc+crypto"
+          },
+          {
+            "versions": "7:",
+            "flags": "-mcpu=thunderx2t99"
+          }
+        ],
+        "clang": {
+          "versions": ":",
+          "flags": "-march=armv8-a -mcpu=generic"
```

Comment:
Was this copied verbatim? Do we need to check the options for `clang`?

Reply:
The setting "-march=armv8xx" seems to be correct on aarch64 machines, but the settings for each version are still under scrutiny. Perhaps you can follow this file to see what you can set.
https://github.com/llvm/llvm-project/blob/master/llvm/include/llvm/Support/AArch64TargetParser.def

Reply:
I think we can have the same values displayed by:
```console
$ llc -march=aarch64 -mcpu=help
```

Reply:
Thanks. I executed the command `llc` in llvm9.0, and following is shown: 
```
[n0013@apollo13 bin]$ ./llc -march=aarch64 -mcpu=help
Available CPUs for this target:

  apple-latest - Select the apple-latest processor.
  cortex-a35   - Select the cortex-a35 processor.
  cortex-a53   - Select the cortex-a53 processor.
  cortex-a55   - Select the cortex-a55 processor.
  cortex-a57   - Select the cortex-a57 processor.
  cortex-a72   - Select the cortex-a72 processor.
  cortex-a73   - Select the cortex-a73 processor.
  cortex-a75   - Select the cortex-a75 processor.
  cortex-a76   - Select the cortex-a76 processor.
  cortex-a76ae - Select the cortex-a76ae processor.
  cyclone      - Select the cyclone processor.
  exynos-m1    - Select the exynos-m1 processor.
  exynos-m2    - Select the exynos-m2 processor.
  exynos-m3    - Select the exynos-m3 processor.
  exynos-m4    - Select the exynos-m4 processor.
  exynos-m5    - Select the exynos-m5 processor.
  falkor       - Select the falkor processor.
  generic      - Select the generic processor.
  kryo         - Select the kryo processor.
  saphira      - Select the saphira processor.
  thunderx     - Select the thunderx processor.
  thunderx2t99 - Select the thunderx2t99 processor.
  thunderxt81  - Select the thunderxt81 processor.
  thunderxt83  - Select the thunderxt83 processor.
  thunderxt88  - Select the thunderxt88 processor.
  tsv110       - Select the tsv110 processor.

Available features for this target:
  many features...

Use +feature to enable a feature, or -feature to disable it.
For example, llc -mcpu=mycpu -mattr=+feature1,-feature2
[n0013@apollo13 bin]$
```
If the corresponding CPU can be specified, it seems good to use the `-mcpu` option. In this case, I will set `-mcpu=thunderx2t99`.